packages/styles/schematics/utils/css/css-migration.ts

```diff
@@ -15,16 +15,16 @@ export class CssMigration {
 
     currentElement (element: TmplAstElement | null) {
         this.filteredUpdates = this.updates
-            .filter(u => !u.tagSelector || u.tagSelector === element?.name)
-            .filter(u => !u.classSelector || element?.attributes.filter(a => a.name === 'class' && a.value.split(' ').includes(u.classSelector)))
-            .filter(u => !u.attributeSelector || element?.attributes.some(a => a.name === u.attributeSelector));
+            .filter(u => u.tag === null || (u.tag !== null && u.tag.test(element?.name ?? '')))
+            .filter(u => u.class === null || element?.attributes.some(a => a.name === 'class' && a.value.split(' ').some(v => u.class !== null && u.class.test(v))))
+            .filter(u => u.attribute === null || element?.attributes.some(a => u.attribute !== null && u.attribute.test(a.name)));
     }
 
     currentTemplate (element: TmplAstTemplate | null) {
         this.filteredUpdates = this.updates
-            .filter(u => !u.tagSelector || u.tagSelector === element?.tagName)
-            .filter(u => !u.classSelector || element?.attributes.filter(a => a.name === 'class' && a.value.split(' ').includes(u.classSelector)))
-            .filter(u => !u.attributeSelector || element?.attributes.some(a => a.name === u.attributeSelector));
+            .filter(u => u.tag === null || (u.tag !== null && u.tag.test(element?.tagName ?? '')))
```

Comment:
This could be simplified: 
`u.tag === null || u.tag.test(element?.tagName ?? '')`

Same for the two other filters

Reply:
Can you provide an simplified example of this line?

Reply:
```suggestion
            .filter(u => u.tag === null || u.tag.test(element?.tagName ?? ''))
```
Either the tag selector is null or the test should be true